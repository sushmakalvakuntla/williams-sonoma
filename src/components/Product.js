import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { Image, Modal } from 'react-bootstrap';
import './Product.scss';
import CarouselComponent from './CarouselComponent';

const Product = ({ product }) => {
  const [show, setShow] = useState(false);
  const handleClose = () => setShow(false);

  return (
    <div
      style={{
        position: 'relative',
        color: 'black',
        marginBottom: '20px',
      }}
    >
      <Image
        src={product.hero.href}
        style={{
          width: `${product.hero.width}`,
          height: `${product.hero.height}`,
        }}
        onClick={() => setShow(true)}
      />
      <div className="product-name">{`${product?.name} `}</div>
      <div className="product-price">
        {`$${
          product.price
            ? product.price.regular
            : product.priceRange.selling.high
          }`}
      </div>
      <Modal show={show} onHide={handleClose}>
        <Modal.Header closeButton>
          <CarouselComponent images={product?.images} />
        </Modal.Header>
      </Modal>
    </div>
  );
};

Product.propTypes = {
  product: PropTypes.object.isRequired,
};

export default Product;
