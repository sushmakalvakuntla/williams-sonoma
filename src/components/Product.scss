.product-price {
  position: absolute;
  bottom: 20px;
  height: 20px;
  width: auto;
  left: 16px;
  background: grey;
  color: white;
  font-weight: 700;
}

.product-name {
  position: absolute;
  top: 1px;
  width: 362px;
  height: 40px;
  left: 1px;
  background-color: rgba(255, 255, 255, 0.6);
  color: black;
  font-weight: 700;
  text-align: left;
  padding: 10px ;
}
