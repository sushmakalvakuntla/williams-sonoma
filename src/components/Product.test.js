import React from 'react';
import { mount } from 'enzyme';
import Product from './Product';
import { Modal, Image } from 'react-bootstrap';

describe('Product', () => {
  const product = {
    id: 'festive-stripe-salad-plate-d6827',
    name: 'Festive Stripe Salad Plate',
    links: {
      www: 'https://www.westelm.com/products/festive-stripe-salad-plate-d6827/',
    },
    price: {
      regular: 10.5,
      selling: 10.5,
    },
    hero: {
      size: 'm',
      meta: '',
      alt: '',
      rel: 'hero',
      width: 363,
      href:
        'https://assets.weimgs.com/weimgs/ab/images/wcm/products/202041/0022/festive-stripe-salad-plate-m.jpg',
      height: 363,
    },
    images: [
      {
        size: "m",
        meta: "",
        alt: "",
        rel: "althero",
        width: 363,
        href: "https://assets.weimgs.com/weimgs/ab/images/wcm/products/202111/0061/walnut-led-desk-lamp-2-m.jpg",
        height: 363
      },
    ],
  };

  let modalShow = false;
  let wrapper;
  beforeEach(() => {
    const container = document.createElement("div");
    document.body.appendChild(container);
    wrapper = mount(<Product show={modalShow} product={product} />);
  });

  it('renders Product correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('renders Image component', () => {
    expect(wrapper.find(Image).length).toEqual(1);
  });

  it('renders Modal component on clicking image', () => {
    expect(wrapper.find(Modal).prop('show')).toBe(false);

    wrapper.find('Image').simulate('click');
    expect(wrapper.find(Modal).prop('show')).toBe(true);
  });

  it('renders major html elements', () => {
    expect(wrapper.find('.product-name').length).toEqual(1);
    expect(wrapper.find('.product-price').length).toEqual(1);
  });
});