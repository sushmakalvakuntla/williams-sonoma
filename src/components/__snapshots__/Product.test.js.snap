// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product renders Product correctly 1`] = `
<Product
  product={
    Object {
      "hero": Object {
        "alt": "",
        "height": 363,
        "href": "https://assets.weimgs.com/weimgs/ab/images/wcm/products/202041/0022/festive-stripe-salad-plate-m.jpg",
        "meta": "",
        "rel": "hero",
        "size": "m",
        "width": 363,
      },
      "id": "festive-stripe-salad-plate-d6827",
      "images": Array [
        Object {
          "alt": "",
          "height": 363,
          "href": "https://assets.weimgs.com/weimgs/ab/images/wcm/products/202111/0061/walnut-led-desk-lamp-2-m.jpg",
          "meta": "",
          "rel": "althero",
          "size": "m",
          "width": 363,
        },
      ],
      "links": Object {
        "www": "https://www.westelm.com/products/festive-stripe-salad-plate-d6827/",
      },
      "name": "Festive Stripe Salad Plate",
      "price": Object {
        "regular": 10.5,
        "selling": 10.5,
      },
    }
  }
  show={false}
>
  <div
    style={
      Object {
        "color": "black",
        "marginBottom": "20px",
        "position": "relative",
      }
    }
  >
    <Image
      fluid={false}
      onClick={[Function]}
      rounded={false}
      roundedCircle={false}
      src="https://assets.weimgs.com/weimgs/ab/images/wcm/products/202041/0022/festive-stripe-salad-plate-m.jpg"
      style={
        Object {
          "height": "363",
          "width": "363",
        }
      }
      thumbnail={false}
    >
      <img
        className=""
        onClick={[Function]}
        src="https://assets.weimgs.com/weimgs/ab/images/wcm/products/202041/0022/festive-stripe-salad-plate-m.jpg"
        style={
          Object {
            "height": "363",
            "width": "363",
          }
        }
      />
    </Image>
    <div
      className="product-name"
    >
      Festive Stripe Salad Plate 
    </div>
    <div
      className="product-price"
    >
      $10.5
    </div>
    <Modal
      animation={true}
      autoFocus={true}
      backdrop={true}
      dialogAs={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "displayName": "ModalDialog",
          "render": [Function],
        }
      }
      enforceFocus={true}
      keyboard={true}
      onHide={[Function]}
      restoreFocus={true}
      show={false}
    >
      <Modal
        autoFocus={true}
        backdrop={true}
        backdropTransition={[Function]}
        containerClassName="modal-open"
        enforceFocus={true}
        keyboard={true}
        manager={
          BootstrapModalManager {
            "containers": Array [],
            "data": Array [],
            "handleContainerOverflow": true,
            "hideSiblingNodes": true,
            "modals": Array [],
            "scrollbarSize": 0,
          }
        }
        onEnter={[Function]}
        onEntering={[Function]}
        onEscapeKeyDown={[Function]}
        onExit={[Function]}
        onExited={[Function]}
        onHide={[Function]}
        renderBackdrop={[Function]}
        renderDialog={[Function]}
        restoreFocus={true}
        show={false}
        transition={[Function]}
      />
    </Modal>
  </div>
</Product>
`;
